File: 74-search-a-2d-matrix/74-search-a-2d-matrix.java
class Solution {
    public boolean searchMatrix(int[][] m, int t) {
        int r = m.length;
        int c = m[0].length;
        int start = 0,end = r*c-1;
        while(start<=end){
            int mid = start + (end - start)/2;
            int val = m[mid/c][mid%c];
            if(val == t) return true;
            else if(val<t) start = mid + 1;
            else end = mid - 1;
        }
        return false;
    }
}

File: 74-search-a-2d-matrix/74-search-a-2d-matrix.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& v, int t) {
        int r = 0,c = 0;
        for(int i = 0;i<v.size();i++){
            if(t>=v[i][0]&&t<=v[i][v[i].size()-1]){
                r = i;
                c = v[i].size();
                break;
            }
        }
        // cout<<r<<" "<<c;
        bool ans = false;
        for(int i = 0 ; i<v[r].size();i++){
            if(v[r][i]==t){
                ans = true;
            }
        }
        return ans;
    }
};